cmake/CPM.cmake

```diff
@@ -0,0 +1,505 @@
+# CPM.cmake - CMake's missing package manager
```

Comment:
Is it not possible to get this file via FetchContent so we don't have to vendor it into RMM directly?

Reply:
CPM actually does recommend  to use this as `CPM.cmake`: https://github.com/TheLartians/CPM.cmake/blob/master/cmake/get_cpm.cmake, which will automatically download CPM. 
I instead copied in the whole file because of the comments [here](https://github.com/rapidsai/cudf/pull/6095#discussion_r477643040), but personally I don't mind either way.

Reply:
I think they are suggesting that we create a RAPIDS fork of the file, although I think as long as we have a way to get a specific tag/version that won't change underneath us we should be OK. I definitely prefer either forking (which one of the RAPIDS org would need to do) or fetching a specific version to vendoring the file into RMM.   We are able to fetch specific versions of other dependencies without vendoring, I don't see why CPM whould be any different.

Reply:
I updated the PR to use their `get_cpm.cmake` scripts used as `CPM.cmake`, which will automatically download the
real `CPM.cmake` (from a specified release) and include it. That also supports caching on the local machine, ie, things can work in offline mode if `CPM_SOURCE_CACHE` is used.